cpp/src_prims/metrics/adjustedRandIndex.h

```diff
@@ -1,158 +1,132 @@
-  //device variables
-  MLCommon::device_buffer<int> a(allocator, stream, numUniqueClasses);
-  MLCommon::device_buffer<int> b(allocator, stream, numUniqueClasses);
-  MLCommon::device_buffer<int> d_aCTwoSum(allocator, stream, 1);
-  MLCommon::device_buffer<int> d_bCTwoSum(allocator, stream, 1);
-  MLCommon::device_buffer<int> d_nChooseTwoSum(allocator, stream, 1);
-  //host variables
-  int h_aCTwoSum;
-  int h_bCTwoSum;
-  int h_nChooseTwoSum;
-
-  //initializing device memory
-  CUDA_CHECK(
-    cudaMemsetAsync(a.data(), 0, numUniqueClasses * sizeof(int), stream));
-  CUDA_CHECK(
-    cudaMemsetAsync(b.data(), 0, numUniqueClasses * sizeof(int), stream));
-  CUDA_CHECK(cudaMemsetAsync(d_aCTwoSum.data(), 0, sizeof(int), stream));
-  CUDA_CHECK(cudaMemsetAsync(d_bCTwoSum.data(), 0, sizeof(int), stream));
-  CUDA_CHECK(cudaMemsetAsync(d_nChooseTwoSum.data(), 0, sizeof(int), stream));
-
-  //calculating the sum of NijC2
-  MLCommon::LinAlg::mapThenSumReduce<int, nCTwo<int>>(
-    d_nChooseTwoSum.data(), numUniqueClasses * numUniqueClasses, nCTwo<int>(),
-    stream, dContingencyMatrix.data(), dContingencyMatrix.data());
-
-  //calculating the row-wise sums
-  MLCommon::LinAlg::reduce<int, int, int>(a.data(), dContingencyMatrix.data(),
-                                          numUniqueClasses, numUniqueClasses, 0,
-                                          true, true, stream);
-
-  //calculating the column-wise sums
-  MLCommon::LinAlg::reduce<int, int, int>(b.data(), dContingencyMatrix.data(),
-                                          numUniqueClasses, numUniqueClasses, 0,
-                                          true, false, stream);
-
-  //calculating the sum of number of unordered pairs for every element in a
-  MLCommon::LinAlg::mapThenSumReduce<int, nCTwo<int>>(
-    d_aCTwoSum.data(), numUniqueClasses, nCTwo<int>(), stream, a.data(),
-    a.data());
-
-  //calculating the sum of number of unordered pairs for every element of b
-  MLCommon::LinAlg::mapThenSumReduce<int, nCTwo<int>>(
-    d_bCTwoSum.data(), numUniqueClasses, nCTwo<int>(), stream, b.data(),
-    b.data());
-
-  //updating in the host memory
-  MLCommon::updateHost(&h_nChooseTwoSum, d_nChooseTwoSum.data(), 1, stream);
-  MLCommon::updateHost(&h_aCTwoSum, d_aCTwoSum.data(), 1, stream);
-  MLCommon::updateHost(&h_bCTwoSum, d_bCTwoSum.data(), 1, stream);
-
-  //freeing the memories in the device
-  if (pWorkspace) CUDA_CHECK(cudaFree(pWorkspace));
-
-  //calculating the ARI
-  int nChooseTwo = ((size) * (size - 1)) / 2;
-  double expectedIndex =
-    ((double)((h_aCTwoSum) * (h_bCTwoSum))) / ((double)(nChooseTwo));
-  double maxIndex = ((double)(h_bCTwoSum + h_aCTwoSum)) / 2.0;
-  double index = (double)h_nChooseTwoSum;
-
-  //checking if the denominator is zero
-  if (maxIndex - expectedIndex)
-    return (index - expectedIndex) / (maxIndex - expectedIndex);
-  else
-    return 0;
-}
-
-};  //end namespace Metrics
-};  //end namespace MLCommon
+/*
```

Comment:
The reason this whole file shows up as a diff is because the previous version of the file had windows CRLF endings! Not sure how it happened, though. Sorry about the trouble caused during reviewing of this file.